crates/storage/provider/src/providers/static_file/manager.rs

```diff
@@ -729,12 +711,13 @@ impl<N: NodePrimitives> StaticFileProvider<N> {
         segment: StaticFileSegment,
         tx: u64,
     ) -> Option<SegmentRangeInclusive> {
-        let static_files = self.static_files_tx_index.read();
-        let segment_static_files = static_files.get(&segment)?;
+        let indexes = self.indexes.read();
```

Comment:
do we need to hold the read lock for the entire fn?

Reply:
we do, because the ranges from this index https://github.com/paradigmxyz/reth/blob/90abe8dc9dd5159bb80a9db30381788541090c9e/crates/storage/provider/src/providers/static_file/manager.rs#L716-L720 should be in sync with this index https://github.com/paradigmxyz/reth/blob/90abe8dc9dd5159bb80a9db30381788541090c9e/crates/storage/provider/src/providers/static_file/manager.rs#L731